cmd/ctr/utils.go

```diff
@@ -112,6 +118,43 @@ func getExecutionService(context *cli.Context) (execution.ContainerServiceClient
 	return execution.NewContainerServiceClient(conn), nil
 }
 
+func getContentProvider(context *cli.Context) (content.Provider, error) {
+	conn, err := getGRPCConnection(context)
+	if err != nil {
+		return nil, err
+	}
+	return contentservice.NewProviderFromClient(contentapi.NewContentClient(conn)), nil
+}
+
+func getRootFSService(context *cli.Context) (rootfsapi.RootFSClient, error) {
+	conn, err := getGRPCConnection(context)
+	if err != nil {
+		return nil, err
+	}
+	return rootfsapi.NewRootFSClient(conn), nil
+}
+
+func getDB(ctx *cli.Context, readonly bool) (*bolt.DB, error) {
+	// TODO(stevvooe): For now, we operate directly on the database. We will
```

Comment:
@samuelkarp The goal is to allow containerd to control the contents of the metadata in a way that may require a daemon. This might be lock coordination or garbage collection but these implementations will define the exact requirements.